### ISSUE TITLE
Stencil in project config causing SQL error

### ISSUE THREAD

**Issue by joshuabaker** [verbb/formie#543]
**Description**

Getting an error back from MySQL related to stencils when running `project-config/apply`.

```
error: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`dandaraliving`.`formie_stencils`, CONSTRAINT `fk_jlppxvjvgnsmrsajnmcjedsqpcilrjmqxexe` FOREIGN KEY (`defaultStatusId`) REFERENCES `formie_statuses` (`id`) ON DELETE SET NULL)
```

Removing the stencil from the project config fixes the issue.

Rebuilding the project config from a backup results in the same issue.

**Additional info**
- Plugin version: 1.4.13
- Craft version: Craft Pro 3.7.8
- Multi-site: Yes

**Comment by engram-design** [comment]
Are you able to report on what form statuses you have? And also sharing your stencil from `config/project/formie/stencils`.

It sounds like it's trying to reference a form status that doesn't exist.

**Comment by joshuabaker** [comment]
Nobody’s changed the statuses. We just have “New”, which I believe is the default.

![image](https://user-images.githubusercontent.com/160484/129716843-3e6b4c6c-a9f6-443e-b448-2eb808feef64.png)

Either way, rebuilding the project config shouldn’t allow reference to a non-existent status. Looking now, the `defaultStatusId` in the YAML is set to `0`.

**Comment by engram-design** [comment]
Is it a custom stencil made by yourself, or the default "Contact Form" one? I can't immediately replicate it, but will do some more testing.

**Comment by joshuabaker** [comment]
Just the default setup.

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.14",
  "...": "..."
}
```

Then run `composer update`. I'll say I think this issue comes down to how project config values might've been stored pre Craft 3.6. The value should never be `0`, but `null` is acceptable. It might've been the lack of proper typecasting in previous Craft versions.

**Comment by engram-design** [comment]
Fixed in [1.4.15](https://github.com/verbb/formie/releases/tag/1.4.15)
